crates/btc-relay/src/lib.rs

```diff
@@ -535,6 +539,7 @@ impl<T: Config> Pallet<T> {
                 // if we added a block to a fork, we may need to reorder the chains
                 Self::reorganize_chains(&blockchain)?;
             } else {
+                Self::update_no_ongoing_fork(current_block_height)?;
```

Comment:
I think this code path can be taken even when there is a fork ahead of the mainchain (since I think reordering is not done until a fork gets 6 blocks ahead). If that is the case, this `update_no_ongoing_fork` is incorrect.

This PR _looks_ like a simple change, but the bitcoin code is deceptively complex and error prone. I would like to see more tests, including for the scenario above. Additionally, I would like to have a check that the error status is equal to the output of `ensure_no_ongoing_fork` in `check_store_block_header_invariants`